### PR TITLE
[MAR-2532] Validate kind directories directly

### PR DESCRIPTION
**Pull request by isaachinman**
## Human summary

Validation now catches invalid or colliding Encephalon kind folders even when those folders are empty.

## Summary

- Validate every non-reserved kind directory with the same portable kind predicate used for record envelopes.
- Report stable directory-level issues for invalid kind names and case/Unicode-normalization collisions.
- Preserve existing record path mismatch checks for records inside scanned directories.
- Add cross-platform tests for portable kind predicates, empty invalid directories, host-supported case/normalization collisions, invalid directories containing records, and valid empty custom kinds.

### CODE REVIEW

**Comment by isaachinman** [comment]
@greptileai review

**Comment by isaachinman** [comment]
@greptileai review

**Comment by isaachinman** [comment]
@pullfrog

**Comment by isaachinman** [comment]
@pullfrog please re-review after main merge.

**Comment by isaachinman** [comment]
@pullfrog review

File: test/records.test.ts [hunk 1]
```diff
@@ -217,6 +227,81 @@ describe('canonical records', () => {
...
+    const unicodeNames = ['cafe\u0301', 'café'].filter(name => canCreateDirectory(root, name))
+    const expected = [
+      ...(caseVariantCreated
+        ? [
+            ['INVALID_KIND_DIRECTORY', 'encephalon/Context'],
+            ['KIND_DIRECTORY_COLLISION', 'encephalon/Context'],
+          ]
+        : []),
+      ...unicodeNames.map(name => ['INVALID_KIND_DIRECTORY', `encephalon/${name}`]),
+      ...(unicodeNames.length === 2 ? [['KIND_DIRECTORY_COLLISION', 'encephalon/café']] : []),
```
[the remark this answers is not in the document]
Comment (isaachinman) [reply test/records.test.ts]: Fixed in 617ecf6. The test no longer hardcodes `encephalon/café`; it computes the expected Unicode collision path from the actual `encephalon` directory entries after applying the same `localeCompare` ordering used by the scanner, then flags whichever canonically equivalent name is encountered second. I also saved the lesson in Encephalon as `review.locale-collision-tests`.

File: test/records.test.ts [hunk 2]
```diff
@@ -260,6 +270,95 @@ describe('canonical records', () => {
...
+            ['KIND_DIRECTORY_COLLISION', 'encephalon/Context'],
+          ]
+        : []),
+      ...unicodeNames.map(name => ['INVALID_KIND_DIRECTORY', `encephalon/${name}`]),
+      ...unicodeCollisionPaths.map(path => ['KIND_DIRECTORY_COLLISION', path]),
+    ].sort((first, second) => String(first[1]).localeCompare(String(second[1])))
+
+    const result = api.validateRecords({ root })
+    assert.equal(result.valid, expected.length === 0)
+    assert.deepEqual(
```
[the remark this answers is not in the document]
Comment (isaachinman) [reply test/records.test.ts:328]: Fixed in ebc3d36. The test now builds `unicodeOrderedNames` from the actual `encephalon` directory listing after applying the scanner’s `localeCompare` ordering, then uses that same ordered list for both `INVALID_KIND_DIRECTORY` and `KIND_DIRECTORY_COLLISION` expectations. That removes the remaining literal-order dependency Pullfrog identified.
